mirgecom/steppers.py

```diff
@@ -82,21 +110,28 @@ def _advance_state_stepper_func(rhs, timestepper,
         the current time
     state: numpy.ndarray
     """
+    t = np.float64(t)
+
     if t_final <= t:
         return istep, t, state
 
+    actx = state.array_context
+    compiled_rhs = compile_rhs(actx, rhs, state)
+
     while t < t_final:
+        state = thaw(freeze(state, actx), actx)
 
         if logmgr:
             logmgr.tick_before()
 
         if get_timestep:
-            dt = get_timestep(state=state, t=t, dt=dt)
+            dt = np.float64(get_timestep(state=state, t=t, dt=dt))
```

Comment:
`get_timestep` is no longer used in mirgecom.  I am curious, why this?  It seems this may be related to the issue of needing to tweak the finishing tolerance.

Reply:
On current main it does use `get_timestep` https://github.com/illinois-ceesd/mirgecom/blob/6359209f134add90c051bd8c94c58470f76c4570/mirgecom/steppers.py#L93-L94 Probably for back compat? Over here, we just cast the dtype to float64.

Reply:
This remains just for backwards compat - but all of the examples and the production drivers have been updated to not use it, and added this to the stepper interface in mirgecom@main:
```python
    if get_timestep is not None:
        from warnings import warn
        warn("Passing the get_timestep function into the stepper is deprecated. "
             "Users should use the dt argument for constant timestep, and "
             "perform any dt modification in the {pre,post}-step callbacks.",
             DeprecationWarning, stacklevel=2)
```

Reply:
I'm still curious about why this casting is needed, anyway.  There are lots of functions that return a number, why does this one in-particular require this casting?

Reply:
This is a slight incompetence on the PytatoArrayContext implementation. Should be a small fix. Will resolve this once fixed.

Reply:
Will remove this line after https://github.com/inducer/arraycontext/pull/65 goes in! (Thanks for catching that! :D)

Reply:
Let's add this as a comment and merge this PR before  inducer/arraycontext#65 goes in?